### PR TITLE
applications: nrf_desktop: Reduce MCUboot memory footprint

### PR DESCRIPTION
**Pull request by MarekPieta**
Change disables MCUboot's logs over RTT to reduce MCUboot memory footprint on nrf52840dk_nrf52840 in "prj_mcuboot_qspi.conf" configuration. This is needed to avoid FLASH overflows.

Jira: NCSDK-24735

### CODE REVIEW

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -201,6 +201,7 @@ nRF Desktop
     The option can be enabled even when the Bluetooth controller is not enabled as part of the application that uses :ref:`nrf_desktop_ble_latency`.
   * The :ref:`nrf_desktop_usb_state` to use :c:func:`usb_hid_set_proto_code` function to set the HID Boot Interface protocol code.
     The ``CONFIG_USB_HID_BOOT_PROTOCOL`` Kconfig option was removed and dedicated API needs to be used instead.
+  * Disabled MCUboot's logs over RTT (:kconfig:option:`CONFIG_LOG_BACKEND_RTT` and :kconfig:option:`CONFIG_USE_SEGGER_RTT`) on nrf52840dk_nrf52840 in ``prj_mcuboot_qspi.conf`` configuration to reduce MCUboot memory footprint and avoid FLASH overflows.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]:
```suggestion
  * Disabled MCUboot's logs over RTT (:kconfig:option:`CONFIG_LOG_BACKEND_RTT` and :kconfig:option:`CONFIG_USE_SEGGER_RTT`) on nrf52840dk_nrf52840 in ``prj_mcuboot_qspi.conf`` configuration to reduce MCUboot memory footprint and avoid flash overflows.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -201,6 +201,7 @@ nRF Desktop
     The option can be enabled even when the Bluetooth controller is not enabled as part of the application that uses :ref:`nrf_desktop_ble_latency`.
   * The :ref:`nrf_desktop_usb_state` to use :c:func:`usb_hid_set_proto_code` function to set the HID Boot Interface protocol code.
     The ``CONFIG_USB_HID_BOOT_PROTOCOL`` Kconfig option was removed and dedicated API needs to be used instead.
+  * Disabled MCUboot's logs over RTT (:kconfig:option:`CONFIG_LOG_BACKEND_RTT` and :kconfig:option:`CONFIG_USE_SEGGER_RTT`) on nrf52840dk_nrf52840 in ``prj_mcuboot_qspi.conf`` configuration to reduce MCUboot memory footprint and avoid flash overflows.
```
Comment (divipillai) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]:
```suggestion
  * Disabled MCUboot's logs over RTT (:kconfig:option:`CONFIG_LOG_BACKEND_RTT` and :kconfig:option:`CONFIG_USE_SEGGER_RTT`) on ``nrf52840dk_nrf52840`` in :file:`prj_mcuboot_qspi.conf` configuration to reduce MCUboot memory footprint and avoid flash overflows.
```

**Review by divipillai — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -201,6 +201,7 @@ nRF Desktop
     The option can be enabled even when the Bluetooth controller is not enabled as part of the application that uses :ref:`nrf_desktop_ble_latency`.
   * The :ref:`nrf_desktop_usb_state` to use :c:func:`usb_hid_set_proto_code` function to set the HID Boot Interface protocol code.
     The ``CONFIG_USB_HID_BOOT_PROTOCOL`` Kconfig option was removed and dedicated API needs to be used instead.
+  * Disabled MCUboot's logs over RTT (:kconfig:option:`CONFIG_LOG_BACKEND_RTT` and :kconfig:option:`CONFIG_USE_SEGGER_RTT`) on ``nrf52840dk_nrf52840`` in :file:`prj_mcuboot_qspi.conf` configuration to reduce MCUboot memory footprint and avoid flash overflows.
```
Comment (mkapala-nordic) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:204]: Consider mentioning that UART backend has been enabled instead

**Review by mkapala-nordic — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 3 above)
Comment (MarekPieta) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:204]: The UART backend was enabled all of the time (board enables it) :)

Comment (mkapala-nordic) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:204]: Then why it has been enabled explicitly in configuration if it is not necessary in theory as board enables it?

Comment (MarekPieta) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:204]: Ok, updated the release note :)

**Review by mkapala-nordic — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 3 above)
Comment (mkapala-nordic) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:204]: Resolved

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -201,6 +201,8 @@ nRF Desktop
...
+    Explicitly enabled the UART log backend (:kconfig:option:`CONFIG_LOG_BACKEND_UART`) together with its dependencies in the mentioned configuration to ensure logs visibility.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]:
```suggestion
    Explicitly enabled the UART log backend (:kconfig:option:`CONFIG_LOG_BACKEND_UART`) together with its dependencies in the configuration file to ensure log visibility.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

**Review by kapi-no — APPROVED** [review]

**Comment by MarekPieta** [comment]
Fix is already cherry-picked to the upmerge branch. I close the PR.

**Pull request closed without merging**